src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -173,6 +173,16 @@ void processMembers(IEnumerable<ISymbol> members)
                             {
                                 memberSet = memberSet ?? new HashSet<ISymbol>();
                                 memberSet.Add(member);
+
+                                // Ensure that we include symbols for both parts of partial methods.
+                                if (member is IMethodSymbol method)
+                                {
+                                    var otherPartOfPartialOpt = method.PartialDefinitionPart ?? method.PartialImplementationPart;
+                                    if (otherPartOfPartialOpt != null)
+                                    {
+                                        memberSet.Add(otherPartOfPartialOpt);
+                                    }
```

Comment:
1. ick.  too bad this just isn't in the members list.  but i can see the problems that would cause as well.  
2. do we not have an AddIfNotNull extension here?  that might clean things up.

Reply:
> do we not have an AddIfNotNull extension here

I did not find any such existing extension method on HashSet.